easytier/src/instance/instance.rs

```diff
@@ -532,24 +532,34 @@ impl Instance {
 
         if !self.global_ctx.config.get_flags().no_tun {
             #[cfg(not(target_os = "android"))]
-            if let Some(ipv4_addr) = self.global_ctx.get_ipv4() {
-                let mut new_nic_ctx = NicCtx::new(
-                    self.global_ctx.clone(),
-                    &self.peer_manager,
-                    self.peer_packet_receiver.clone(),
-                );
-                new_nic_ctx.run(ipv4_addr).await?;
-                let ifname = new_nic_ctx.ifname().await;
-                Self::use_new_nic_ctx(
-                    self.nic_ctx.clone(),
-                    new_nic_ctx,
-                    Self::create_magic_dns_runner(
-                        self.peer_manager.clone(),
-                        ifname,
-                        ipv4_addr.clone(),
-                    ),
-                )
-                .await;
+            {
+                let ipv4_addr = self.global_ctx.get_ipv4();
+                let ipv6_addr = self.global_ctx.get_ipv6();
+                
+                // Only run if we have at least one IP address (IPv4 or IPv6)
+                if ipv4_addr.is_some() || ipv6_addr.is_some() {
+                    let mut new_nic_ctx = NicCtx::new(
+                        self.global_ctx.clone(),
+                        &self.peer_manager,
+                        self.peer_packet_receiver.clone(),
+                    );
+                    
+                    new_nic_ctx.run(ipv4_addr, ipv6_addr).await?;
+                    let ifname = new_nic_ctx.ifname().await;
+                    
+                    // For Magic DNS, use IPv4 if available, otherwise use a default
+                    let dns_ipv4 = ipv4_addr.unwrap_or_else(|| "169.254.1.1/32".parse().unwrap());
```

Comment:
The magic dns will not work if a ipv4 address not assigned to tun device.

The magic dns server need transfrom the dst ip of the packet sent to magic dns server (100.100.100.101) to the tun ip. And the magic dns server listens on tun ip so it can receive the modified packet. If tun ip is not assigned this process will not work.

Reply:
Considering this patch doesn't add AAAA record support for magic dns server, it's ok not run the magic dns server if ipv4 address is not provided.